torch/distributed/device_mesh.py

```diff
@@ -113,6 +113,41 @@ def create_sub_mesh(
 
             return res_submesh
 
+        def create_flatten_mesh(self, device_mesh: "DeviceMesh") -> "DeviceMesh":
```

Comment:
This is similar to `create_sub_mesh`. The main difference is we need to sort the `mesh_dim_names` to flatten, get the mesh size of the flatten dim, and _init_backend needs to be True. 

We can probably see if we can merge this into `create_sub_mesh` later but want to have this out so we can't unblock the dp cp use case.

Reply:
Do we want to let user to pass the name as we discussed before? Sometimes reusing the existing dim names can be confusing.